kind/deploy.sh

```diff
@@ -0,0 +1,753 @@
+    log_message "1" "   _____                                  _                        "
+    log_message "1" "  / ____|                                | |                       "
+    log_message "1" " | (___    _ __     ___   __      __   __| |  _ __    ___    _ __  "
+    log_message "1" "  \___ \  | '_ \   / _ \  \ \ /\ / /  / _  | |  __|  / _ \  | \ _ \ "
+    log_message "1" "  ____) | | | | | | (_) |  \ V  V /  | (_| | | |    | (_) | | |_) |"
+    log_message "1" " |_____/  |_| |_|  \___/    \_/\_/    \__,_| |_|     \___/  |  __/ "
+    log_message "1" "                                                            | |    "
+    log_message "1" "                                                            |_|    "
+    log_message "1" " Kind installation script"
+    log_message "1" ""
+    log_message "1" "- Deploying a local secured (using htpasswd) docker registry"
+    log_message "1" "- Generating a selfsigned certificate (using openssl) to expose the registry as a HTTP/HTTPS endpoint"
+    log_message "1" "- Setting a docker network between the containers: kind and registry and alias \"registry.local\""
+    log_message "1" "- Allowing to access the repository using as address \"registry.local:${REGISTRY_PORT}\" within a pod, from laptop or when a pod is created"
+    log_message "1" "- Exposing 2 additional NodePort: 30000 and 31000"
+    log_message "1" "- Deploying an ingress controller"
+    log_message "1" "- Copying the generated certificate here: $HOME/local-registry.crt"
+    log_message "1" ""
+    log_message "1" ""
+    note "5" "Variables used:"
+    note "5" ""
+    note "5" "CLUSTER_NAME: ${CLUSTER_NAME}"
+    note "5" "DELETE_KIND_CLUSTER: ${DELETE_KIND_CLUSTER}"
+    note "5" "INGRESS: ${INGRESS}"
+    note "5" "KNATIVE_VERSION: ${KNATIVE_VERSION}"
+    note "5" "KUBERNETES_VERSION: ${KUBERNETES_VERSION}"
+    note "5" "LOGGING_VERBOSITY: ${LOGGING_VERBOSITY}"
+    note "5" "REGISTRY_IMAGE_VERSION: ${REGISTRY_IMAGE_VERSION}"
+    note "5" "REGISTRY_PASSWORD: ${REGISTRY_PASSWORD}"
+    note "5" "REGISTRY_PORT: ${REGISTRY_PORT}"
+    note "5" "REGISTRY_USER: ${REGISTRY_USER}"
+    note "5" "SECURE_REGISTRY: ${SECURE_REGISTRY}"
+    note "5" "SERVER_IP: ${SERVER_IP}"
+    note "5" "SHOW_HELP: ${SHOW_HELP}"
+    note "5" "USE_EXISTING_CLUSTER: ${USE_EXISTING_CLUSTER}"
+}
+
+show_usage() {
+    log_message "0" ""
+    log_message "0" "Usage: "
+    log_message "0" "\t./deploy-kind.sh command [parameters,...]"
+    log_message "0" ""
+    log_message "0" "Available commands: "
+    log_message "0" "\tinstall:\t\t\t\tInstall the kind cluster"
+    log_message "0" "\tremove:\t\t\t\t\tRemove the kind cluster"
+    log_message "0" ""
+    log_message "0" "Required parameters: "
+    log_message "0" "\t--ingress [nginx,kourier]:\t\tIngress to be deployed. One of nginx,kourier."
+    log_message "0" ""
+    log_message "0" "Optional parameters: "
+    log_message "0" "\t-h, --help:\t\t\t\tThis help message"
+    log_message "0" ""
```

Comment:
I dont see a parameter to specify the version to be installed of the k8s cluster as this is the case today. According to the kind tool version installed, they support different k8s versions -see: https://github.com/kubernetes-sigs/kind/releases/tag/v0.18.0

Reply:
The `--kubernetes-version` parameter should be implemented and available for this.

https://github.com/snowdrop/k8s-infra/pull/279/files#diff-e31912bfaa479f5abf54676480bc184cf0cce935c48b30a92a9cb0c48ac982b0R670

```bash
        --kubernetes-version) KUBERNETES_VERSION="$2"; shift ;;
```

Reply:
LGTM